src/solvers/NelderMead.jl

```diff
@@ -1,6 +1,152 @@
+
+"""
+    NelderMeadSimplex
+
+A simplex for the Nelder-Mead algorithm.
+
+# Constructors
+
+    NelderMeadSimplex(M::AbstractManifold)
+
+Construct a simplex using random points from manifold `M`.
+
+    NelderMeadSimplex(
+        M::AbstractManifold,
+        p,
+        B::AbstractBasis=DefaultOrthonormalBasis();
+        a::Real=0.025,
+        retraction_method::AbstractRetractionMethod=default_retraction_method(M),
+    )
+
+Construct a simplex from a basis `B` with one point being `p` and other points
+constructed by moving by `a` in each principal direction defined by basis `B` of the tangent
+space at point `p` using retraction `retraction_method`. This works similarly to how
+the initial simplex is constructed in the Euclidean Nelder-Mead algorithm.
```

Comment:
...just that the construction here is done in the tangent space at `p` – is what one could add here maybe?

Reply:
Sure, I will add a note.